### DISCUSSION TITLE
Minimal examples for display in Vue

### DISCUSSION THREAD

**Discussion by goeil** [area17/twill#1249] [Q&A]
Hello all,

I try to run my Twill project with VueJS front-end (via InertiaJS).
My main struggle remains with preparing the Json data for client-side.

If other developers use JS front-end (and not blade), would they mind posting some minimal examples here?

Particularly:
- attaching media files URL
- attaching tags
- attaching data from related tables
- reducing related data (e.g. removing the pivot table from data)

I also wonder what the truth is with Repository methods handle*, hydrate*… Do they launch automatically, or must they be launched according to circumstances?

I can do some things, but feel like reinventing the wheel.

Any help would help me a lot.

**Comment by pboivin** [comment]
Hi @goeil,

I don't have much experience with Inertia.js but, as you probably know, passing data to the frontend can be done with the `inertia()` helper, just like the standard `view()`:
```
public function show(Event $event)
{
    return inertia('Event/Show', [
        'event' => $event->only('id', 'title', 'start_date', 'description'),
    ]);
}
```

Does this not cover your use case?

For trigerring actions from your frontend, I think you are definitely on the right track exploring Repositories. You can create and manipulate Twill models directly but going through a Repository will trigger all of the appropriate behavior "hooks". For example:
```
app(EventRepository::class)->create([
    // ...
]);
```

The [`create()`](https://github.com/area17/twill/blob/2.x/src/Repositories/ModuleRepository.php#L225) method will call all of the behavior hooks (e.g. [`afterSave()`](https://github.com/area17/twill/blob/2.x/src/Repositories/ModuleRepository.php#L610) -> [`afterSaveHandleBrowsers()`](https://github.com/area17/twill/blob/2.x/src/Repositories/Behaviors/HandleBrowsers.php#L41)).

Hope this can help!
